// canguru-api/controllers/OrganizationsService.js
'use strict';

exports.createOrganization = function(args, res, next) {
  /**
   * Create organization
   * Create an organization. A name must be provided
   *
   * body OrganizationDTO  (optional)
   * returns OrganizationDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.deleteOrganization = function(args, res, next) {
  /**
   * Delete organization
   * Delete an organization.
   *
   * organizationId String 
   * no response value expected for this operation
   **/
  res.end();
}

exports.getOrganization = function(args, res, next) {
  /**
   * Get organization
   * Get an organization for a given ID
   *
   * organizationId String 
   * returns OrganizationDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.listOrganizations = function(args, res, next) {
  /**
   * List organizations
   * Return a list of organizations
   *
   * returns List
   **/
  var examples = {};
  examples['application/json'] = [ {
  "name" : "aeiou",
  "id" : "aeiou"
} ];
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.patchOrganization = function(args, res, next) {
  /**
   * Patch organization
   * Patch an organization.
   *
   * organizationId String 
   * body OrganizationDTO  (optional)
   * returns OrganizationDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.updateOrganization = function(args, res, next) {
  /**
   * Update organization
   * Update an organization. ID must be provided in the request body
   *
   * body OrganizationDTO  (optional)
   * returns OrganizationDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}
