// canguru-api/controllers/Organizations.js
'use strict';

var url = require('url');

var Organizations = require('./OrganizationsService');

module.exports.createOrganization = function createOrganization (req, res, next) {
  Organizations.createOrganization(req.swagger.params, res, next);
};

module.exports.deleteOrganization = function deleteOrganization (req, res, next) {
  Organizations.deleteOrganization(req.swagger.params, res, next);
};

module.exports.getOrganization = function getOrganization (req, res, next) {
  Organizations.getOrganization(req.swagger.params, res, next);
};

module.exports.listOrganizations = function listOrganizations (req, res, next) {
  Organizations.listOrganizations(req.swagger.params, res, next);
};

module.exports.patchOrganization = function patchOrganization (req, res, next) {
  Organizations.patchOrganization(req.swagger.params, res, next);
};

module.exports.updateOrganization = function updateOrganization (req, res, next) {
  Organizations.updateOrganization(req.swagger.params, res, next);
};
