// canguru-api/controllers/Projects.js
'use strict';

var url = require('url');

var Projects = require('./ProjectsService');

module.exports.createActivity = function createActivity (req, res, next) {
  Projects.createActivity(req.swagger.params, res, next);
};

module.exports.createProject = function createProject (req, res, next) {
  Projects.createProject(req.swagger.params, res, next);
};

module.exports.deleteActivity = function deleteActivity (req, res, next) {
  Projects.deleteActivity(req.swagger.params, res, next);
};

module.exports.deleteProject = function deleteProject (req, res, next) {
  Projects.deleteProject(req.swagger.params, res, next);
};

module.exports.getActivity = function getActivity (req, res, next) {
  Projects.getActivity(req.swagger.params, res, next);
};

module.exports.getProject = function getProject (req, res, next) {
  Projects.getProject(req.swagger.params, res, next);
};

module.exports.listProjectActivities = function listProjectActivities (req, res, next) {
  Projects.listProjectActivities(req.swagger.params, res, next);
};

module.exports.listProjects = function listProjects (req, res, next) {
  Projects.listProjects(req.swagger.params, res, next);
};

module.exports.patchActivity = function patchActivity (req, res, next) {
  Projects.patchActivity(req.swagger.params, res, next);
};

module.exports.patchProject = function patchProject (req, res, next) {
  Projects.patchProject(req.swagger.params, res, next);
};

module.exports.updateActivity = function updateActivity (req, res, next) {
  Projects.updateActivity(req.swagger.params, res, next);
};

module.exports.updateProject = function updateProject (req, res, next) {
  Projects.updateProject(req.swagger.params, res, next);
};
