// canguru-api/controllers/ProjectsService.js
'use strict';

exports.createActivity = function(args, res, next) {
  /**
   * Create activity
   * Create an activity. A description must be provided
   *
   * projectId String 
   * body ActivityDTO  (optional)
   * returns ActivityDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "description" : "aeiou",
  "id" : 123456789,
  "projectId" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.createProject = function(args, res, next) {
  /**
   * Create project
   * Create a project. A name must be provided
   *
   * body ProjectDTO  (optional)
   * returns ProjectDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "organizations" : [ {
    "name" : "aeiou",
    "id" : "aeiou"
  } ],
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.deleteActivity = function(args, res, next) {
  /**
   * Delete activity
   * Delete an activity.
   *
   * projectId String 
   * activityId String 
   * no response value expected for this operation
   **/
  res.end();
}

exports.deleteProject = function(args, res, next) {
  /**
   * Delete project
   * Delete a project.
   *
   * projectId String 
   * no response value expected for this operation
   **/
  res.end();
}

exports.getActivity = function(args, res, next) {
  /**
   * Get activity
   * Get an activity for a given ID
   *
   * projectId String 
   * activityId String 
   * returns ActivityDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "description" : "aeiou",
  "id" : 123456789,
  "projectId" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.getProject = function(args, res, next) {
  /**
   * Get project
   * Get a project for a given ID
   *
   * projectId String 
   * returns ProjectDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "organizations" : [ {
    "name" : "aeiou",
    "id" : "aeiou"
  } ],
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.listProjectActivities = function(args, res, next) {
  /**
   * List project activities
   * Return a list of activities for project
   *
   * projectId String 
   * returns List
   **/
  var examples = {};
  examples['application/json'] = [ {
  "name" : "aeiou",
  "description" : "aeiou",
  "id" : 123456789,
  "projectId" : "aeiou"
} ];
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.listProjects = function(args, res, next) {
  /**
   * List projects
   * Return a list of projects
   *
   * returns List
   **/
  var examples = {};
  examples['application/json'] = [ {
  "name" : "aeiou",
  "organizations" : [ {
    "name" : "aeiou",
    "id" : "aeiou"
  } ],
  "id" : "aeiou"
} ];
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.patchActivity = function(args, res, next) {
  /**
   * Patch activity
   * Patch an activity.
   *
   * projectId String 
   * activityId String 
   * body ActivityDTO  (optional)
   * returns ActivityDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "description" : "aeiou",
  "id" : 123456789,
  "projectId" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.patchProject = function(args, res, next) {
  /**
   * Patch project
   * Patch a project.
   *
   * projectId String 
   * body ProjectDTO  (optional)
   * returns ProjectDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "organizations" : [ {
    "name" : "aeiou",
    "id" : "aeiou"
  } ],
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.updateActivity = function(args, res, next) {
  /**
   * Update activity
   * Update an activity. ID must be provided in the request body
   *
   * projectId String 
   * body ActivityDTO  (optional)
   * returns ActivityDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "description" : "aeiou",
  "id" : 123456789,
  "projectId" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}

exports.updateProject = function(args, res, next) {
  /**
   * Update project
   * Update a project. ID must be provided in the request body
   *
   * body ProjectDTO  (optional)
   * returns ProjectDTO
   **/
  var examples = {};
  examples['application/json'] = {
  "name" : "aeiou",
  "organizations" : [ {
    "name" : "aeiou",
    "id" : "aeiou"
  } ],
  "id" : "aeiou"
};
  if (Object.keys(examples).length > 0) {
    res.setHeader('Content-Type', 'application/json');
    res.end(JSON.stringify(examples[Object.keys(examples)[0]] || {}, null, 2));
  } else {
    res.end();
  }
}
